testing/src/test/java/io/stargate/it/http/AuthApiTest.java

```diff
@@ -0,0 +1,181 @@
+import io.stargate.auth.model.Error;
+import io.stargate.auth.model.Secret;
+import io.stargate.auth.model.UsernameCredentials;
+import io.stargate.it.BaseOsgiIntegrationTest;
+import java.io.IOException;
+import net.jcip.annotations.NotThreadSafe;
+import org.apache.http.HttpStatus;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(Parameterized.class)
+@NotThreadSafe
+public class AuthApiTest extends BaseOsgiIntegrationTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(RestApiv2Test.class);
+
+  @Rule public TestName name = new TestName();
+
+  private static String host = "http://" + stargateHost;
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  public void authTokenGenerate() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth/token/generate", host),
+            objectMapper.writeValueAsString(new Secret("cassandra", "cassandra")),
+            HttpStatus.SC_CREATED);
+
+    AuthTokenResponse authTokenResponse = objectMapper.readValue(body, AuthTokenResponse.class);
+    assertThat(authTokenResponse.getAuthToken()).isNotNull();
+  }
+
+  @Test
+  public void authTokenGenerate_BadUsername() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth/token/generate", host),
+            objectMapper.writeValueAsString(new Secret("bad_username", "cassandra")),
+            HttpStatus.SC_UNAUTHORIZED);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription())
+        .isEqualTo(
+            "Failed to create token: Provided username bad_username and/or password are incorrect");
+  }
+
+  @Test
+  public void authTokenGenerate_BadPassword() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth/token/generate", host),
+            objectMapper.writeValueAsString(new Secret("cassandra", "bad_password")),
+            HttpStatus.SC_UNAUTHORIZED);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription())
+        .isEqualTo(
+            "Failed to create token: Provided username cassandra and/or password are incorrect");
+  }
+
+  @Test
+  public void authTokenGenerate_MissingSecret() throws IOException {
+    String body =
+        RestUtils.post(
+            "",
+            String.format("%s:8081/v1/auth/token/generate", host),
+            "",
+            HttpStatus.SC_BAD_REQUEST);
+
+    Error error = objectMapper.readValue(body, Error.class);
+    assertThat(error.getDescription()).isEqualTo("Must provide a body to the request");
+  }
+
+  @Test
+  public void auth() throws IOException {
```

Comment:
`auth` I think that that the test name should be more descriptive, i.e:
`performAuthSuccesfully`.

Reply:
These are just following the names of the endpoints they're calling. In this case `/auth`.